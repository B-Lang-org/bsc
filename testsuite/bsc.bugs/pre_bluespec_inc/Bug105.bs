package Bug105(sysBug105) where

-- compile should fail old Bug 105
-- import Vector

sysBug105 :: Module Empty
sysBug105 =
  let
    l :: Vector 2 (Bit 16) = replicate 1
  in
      module
        finalvals :: Reg (Bit (TMul 2 16))
        finalvals <- mkReg (pack l)

        rules
          when True ==> $display "%0d" (finalvals :: (Bit (TMul 2 16)))

