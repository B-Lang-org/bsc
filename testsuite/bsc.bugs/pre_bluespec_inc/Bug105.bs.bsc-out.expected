checking package dependencies
compiling Bug105.bs
Error: "Bug105.bs", line 9, column 9: (T0007)
  Unbound type constructor `Vector'
